Newer Consul version available shouldn't be an error

#### Overview of the Issue

The `Newer Consul version available` log doesn't need to be an error. It should be a informational message or warning at most. We typically act/alert on error messages and have hundreds of servers firing this.

#### Reproduction Steps

N/A

### Consul info for both Client and Server

N/A

### Operating system and Environment details

CentOS 7 on GCE

### Log Fragments

```
[ERROR] agent: Newer Consul version available: new_version=1.7.2 current_version=1.7.1
```

--- a/command/agent/agent.go
+++ b/command/agent/agent.go
@@ -108,7 +108,7 @@ func (c *cmd) checkpointResults(results *checkpoint.CheckResponse, err error) {
 		return
 	}
 	if results.Outdated {
-		c.logger.Error("Newer Consul version available", "new_version", results.CurrentVersion, "current_version", c.version)
+		c.logger.Info("Newer Consul version available", "new_version", results.CurrentVersion, "current_version", c.version)
 	}
 	for _, alert := range results.Alerts {
 		switch alert.Level {

